### DISCUSSION TITLE
what to do? MS OpenJDK 17.0.11 and 21.0.3 ARM libraries are incompatible to the Prorealtime JAR file version 12.1

### DISCUSSION THREAD

**Discussion by ojutan** [microsoft/openjdk#615] [Q&A] [answered]
Hi, I want to talk about a problem and ask who to contact.
Although the MS OpenJDK is build from public source code it differes from the Oracle version. This is the point for me... I own a Microsoft Surface ProX with ARM CPU and Windows 11 for ARM. And I am using a heavyweight application called Prorealtime Complete, which is  basically a JAR file.
It installs with the Oracle OpenJDK 17.0.11 as 64 bit X64  runtime and the JAR is a heavyweight application that easily eats up 4 or 6 GB... which is quite slow in the emulator.
So I copied  the ARM runtimes into the Java folder of Proreal time... the app launches... and loads forever.  I talked with the vendor but  they said that only very few of there (around one million) users on Windows on ARM. First the support did not understand technically what I wanted from them, then second attempt : "for these 10 users we won'T investigate"

The problem occurs in all binaries of Microsoft OpenJDK, the Intel but also the ARM, 32 and 64 bit.
I think the MS OpenJDK is not completely a clone of the Oracle version and obviously the Oracle runtime  executes the JAr  (but it is intel only). So who to contact?

Of course I could get a newer Surface with 3 times faster CPU that emulates the Intel stuff 3 times faster... but that's waste of ressources as the Intel emulations eats up the battery very fast. So ... how to run a diagnostic on what went wrong? Debugview does not show anything I do understand.

**Comment by karianna** [comment]
@ojutan - Does Prorealtime interact with native components or spin up a Desktop GUI?  Is there a free trial version that we can test with?

**Comment by ojutan** [comment]
hi, sorry for the late answer... the application builds up a GUI in java, but the way it is paunched prohibits any third party testing. 1.) you need to have a contract with the vendor, for later accessing a third-partie's API. You get access mainly by signing up for a contract, get your ID verified and the account approved... then it downloads the app instaler. Once installed the web page launches the app with encrypted parameters having the account information (real or demo)...  I myself cannot look too deep into that software, but my guess is that it initializes its core api until it wants to spawn some child processes to display GUI elements but that never happens. In emulator it comes with several indenendant windows displaying HLOC charts.

Here the vendor Prorealtime has no interest to pursue this issue...  so this might remain unsolved for a longer time. Maybe it happens in "bypass bugfixing" when other issues are solved. For the time being I am using the not-so-bad web interface which works in Chrome for ARM but not in IE Edge Chromium for ARM...

**Reply by karianna** [reply, marked as answer]
OK, best I can offer is to have Prorealtime contact us and we can try to fix it with them.
